### ISSUE TITLE
TypeScript exports support in 4.7

### ISSUE THREAD

**Issue by samhh** [samhh/fp-ts-std#117]
https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#package-json-exports-imports-and-self-referencing

Could improve the manifest configuration, it's a bit hacky at the moment.

**Comment by pasinJ** [comment]
I'm not sure if I should post it here or create a new issue.
In my Typescript project (typescript: 5.0.4, fp-ts-std: 0.16.0), I found that the following error showed up when I enabled the `strict` mode and set `module` to `Node16` or `NodeNext`.

[Reproduction repository](https://github.com/pasinJ/fp-ts-std-import-error)

![fp-ts-std-index](https://user-images.githubusercontent.com/33993773/231395949-4b99d98d-1506-4e3a-a1f3-7d19434ddcef.png)
![fp-ts-std-file](https://user-images.githubusercontent.com/33993773/231395973-9a342de6-c66e-4d50-be7c-60d7239a18a8.png)

According to [Typescript docs](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing),

> With TypeScript’s original Node support, it would look for a "main" field, and then look for declaration files that corresponded to that entry. For example, if "main" pointed to ./lib/index.js, TypeScript would look for a file called ./lib/index.d.ts. A package author could override this by specifying a separate field called "types" (e.g. "types": "./types/index.d.ts").

I think because `fp-ts-std` puts `.d.ts` and `.js` into separated folders, so it may cause Typescript to cannot find the type definitions. If I set the `types` as the document suggested, it can solve the error.

```
fp-ts-std/package.json
...
"exports": {
    ".": {
      "types": "./dist/types/index.d.ts",
      "require": "./dist/cjs/index.js",
      "import": "./dist/esm/index.js"
    },
    "./*": {
      "types": "./dist/types/*.d.ts",
      "require": "./dist/cjs/*.js",
      "import": "./dist/esm/*.js"
    }
  },
...
```

Could we consider applying this to improve the configuration?

**Comment by samhh** [comment]
Released in 0.17.0, looks good on my end.
